app/src/main/java/co/touchlab/droidconandroid/shared/presenter/ConferenceDataHelper.java

```diff
@@ -48,195 +46,201 @@ public static String dateToDayString(Date d)
+                    for(NetworkUserAccount newSpeaker : newEvent.speakers)
+                    {
+                        UserAccount oldSpeaker = helper.getUserAccount(newSpeaker.id);
 
-                        for(co.touchlab.droidconandroid.shared.network.dao.UserAccount ua : event.speakers)
+                        if(oldSpeaker == null)
                         {
-                            UserAccount userAccount = userAccountDao.queryForId(ua.id);
-
-                            if(userAccount == null)
-                            {
-                                userAccount = new UserAccount();
-                            }
-
-                            UserDataHelper.userAccountToDb(ua, userAccount);
-                            userAccountDao.createOrUpdate(userAccount);
-                            Where<EventSpeaker> where = new Where<>(eventSpeakerDao);
-                            List resultList = where.and()
-                                    .eq("event_id", event.id)
-                                    .eq("userAccount_id", userAccount.id)
-                                    .query()
-                                    .list();
-
-                            EventSpeaker eventSpeaker = ((resultList.size() == 0)
-                                    ? new EventSpeaker()
-                                    : (EventSpeaker) resultList.get(0));
-
-                            eventSpeaker.event = event;
-                            eventSpeaker.userAccount = userAccount;
-                            eventSpeaker.displayOrder = speakerCount++;
-
-                            eventSpeakerDao.createOrUpdate(eventSpeaker);
+                            oldSpeaker = new UserAccount();
                         }
-                    }
 
-                }
+                        helper.convertAndSaveUserAccount(newSpeaker, oldSpeaker);
 
-                // don't clear the db if no events are returned
-                if(! foundEvents.isEmpty())
-                {
-                    helper.deleteEventsNotIn(foundEvents);
-                }
+                        EventSpeaker eventSpeaker = helper.getSpeakerForEventWithId(newEvent.id,
+                                newSpeaker.id);
 
-                if(blocks.size() > 0)
-                {
-                    //Dump all of them first
-                    blockDao.delete(blockDao.queryForAll().list());
+                        if(eventSpeaker == null)
+                        {
+                            eventSpeaker = new EventSpeaker();
+                        }
 
-                    for(co.touchlab.droidconandroid.shared.network.dao.Block block : blocks)
-                    {
-                        block.startDateLong = TimeUtils.parseTime(block.startDate);
-                        block.endDateLong = TimeUtils.parseTime(block.endDate);
-                        blockDao.createOrUpdate(block);
+                        // make 1 method, update&save
```

Comment:
im assuming it was too complicated to pass in 3 fields just to consolidate this into 1 method call. If so, we can delete this comment.